x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/CoordinatedInferenceAction.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ml.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.xpack.core.ml.inference.TrainedModelPrefixStrings;
+import org.elasticsearch.xpack.core.ml.inference.trainedmodel.InferenceConfigUpdate;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class CoordinatedInferenceAction extends ActionType<InferModelAction.Response> {
+
+    public static final CoordinatedInferenceAction INSTANCE = new CoordinatedInferenceAction();
+    public static final String NAME = "cluster:internal/xpack/ml/coordinatedinference";
+
+    public CoordinatedInferenceAction() {
+        super(NAME, InferModelAction.Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+
+        public enum ModelType {
+            INFERENCE_SERVICE_MODEL,
+            ML_NODE_PYTORCH_MODEL,
+            BOOSTED_TREE_MODEL,
+            NLP_MODEL,  // Either an inference service model or ml pytorch model but not a boosted tree model
+            UNKNOWN
+        };
+
+        public static Request forInferenceService(String modelId, List<String> inputs, @Nullable Map<String, Object> taskSettings) {
```

Comment:
I'm probably missing something but is the plan to use this from a search context? It doesn't look like we use this outside of tests at the moment.